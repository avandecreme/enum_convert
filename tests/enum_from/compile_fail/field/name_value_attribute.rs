use enum_convert::EnumFrom;

enum Source {
    Struct { x: i32 },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from]
    Struct {
        #[enum_from = "Source::Struct.x"] // Should be #[enum_from(Source::Struct.x)]
        a: i32,
    },
}

fn main() {}
