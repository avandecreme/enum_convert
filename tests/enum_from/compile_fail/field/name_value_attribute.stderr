error: expected a list of field names, for example #[enum_from(Source1::field_a, Source2::field_b)]
  --> tests/enum_from/compile_fail/field/name_value_attribute.rs:14:9
   |
14 |         #[enum_from = "Source::x"] // Should be #[enum_from(Source::x)]
   |         ^
